File: src/test/testdata/inspections/unsafeIterators/IntFloatMap.java
package main;

import com.badlogic.gdx.utils.IntFloatMap;

import java.util.Iterator;

class IntFloatMapTest {

  private IntFloatMap map = new IntFloatMap();

  void test() {

    <warning>map.keys()</warning>.hasNext();
    <warning>map.entries()</warning>.iterator();

    IntFloatMap map2 = map;

    Iterator iterator1 = <warning>map.iterator()</warning>;
    IntFloatMap.Keys keys = <warning>map2.keys()</warning>;
    IntFloatMap.Entries entries = <warning>map2.entries()</warning>;
    IntFloatMap.Values values = <warning>map2.values()</warning>;

    IntFloatMap.Keys keys2 = new IntFloatMap.Keys(map2);

  }

}

File: src/test/testdata/inspections/unsafeIterators/IntFloatMap.kt
package main

import com.badlogic.gdx.utils.IntFloatMap

@Suppress("UNUSED_VARIABLE", "UNCHECKED_CAST")
class IntFloatMapTestKotlin {

  private val map = IntFloatMap()

  companion object {
    var map2: IntFloatMap? = null
  }

  internal fun test() {

    <warning>map.keys()</warning>.hasNext()
    <warning>map.entries()</warning>.iterator()

    map2 = map

    val iterator1 = <warning>map.iterator()</warning>
    val keys = <warning>map2?.keys()</warning>
    val entries = <warning>map2?.entries()</warning>
    val values = <warning>map2?.values()</warning>

    val keys2 = IntFloatMap.Keys(map2)



  }

}

